Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/EditorComponentBase.cpp

```diff
@@ -49,6 +49,17 @@ namespace AzToolsFramework
             m_transform = nullptr;
         }
 
+        void EditorComponentBase::SetEntity(AZ::Entity* entity)
```

Comment:
I've noticed that SetEntity can get called multiple times with the same entity. So, it'd be good to have a 
```
if (m_entity != entity)
```
check wrapping the code in this function too similar to Component::SetEntity()

Reply:
I think it's safer to always call the base class Component::SetEntity() so we don't make assumptions on the base class behavior.